packages/core/src/common.ts

```diff
@@ -0,0 +1,168 @@
+import type {
+  LiteralUnion,
+  SignInOptions,
+  SignInAuthorizationParams,
+  SignOutParams,
+  Session,
+} from "./types"
+import type { BuiltInProviderType, RedirectableProviderType } from "./providers"
+
+interface CreateSignInClientOpts {
+  /**
+   * When no error is returned, call the redirect function. needed because Solid doesn't redirect onSuccess
+   * @default false
+   */
+  redirectWhenNoError?: boolean
+  /**
+   * function that will get called when `res.ok` is true
+   * Nedded to reftech the session onSuccess:
+   * res.ok && await __NEXTAUTH._getSession({ event: "storage" })
+   */
+  onSuccess?: () => void | Promise<void>
+  /**
+   * prefix of the api route
+   */
+  prefix: string
+}
+
+interface CreateSignOutClientOpts {
+  /**
+   * if true, the signout function will redirect after making the request regardless of the response/user config
+   * Needed because SvelteKit redirects in any case
+   * @default false
+   */
+  redirectOnFinish?: boolean
+  /**
+   * function that will get called after the request is made
+   * Nedded to reftech the session:
+   * res.ok && await __NEXTAUTH._getSession({ event: "storage" })
+   */
+  onFinish?: () => void | Promise<void>
+  /**
+   * prefix of the api route
+   */
+  prefix: string
+}
+export function createSignInClient(opts: CreateSignInClientOpts) {
```

Comment:
This should not be part of the core, as clients should be tailored for each framework. In almost no case this is the best way to do this in my opinion, as this uses client-side JS.